packages/block-editor/src/components/page-template-picker/picker.js

```diff
@@ -12,8 +14,17 @@ import getDefaultTemplates from './default-templates';
 import Preview from './preview';
 
 const __experimentalPageTemplatePicker = ( { templates = getDefaultTemplates() } ) => {
+	const { setLayoutTemplate } = useDispatch( 'core/editor' );
 	const [ templatePreview, setTemplatePreview ] = useState();
 
+	const onApply = ( ) => {
+		setLayoutTemplate( {
+			name: templatePreview.name,
+			content: parse( templatePreview.content ),
+		} );
+		setTemplatePreview( undefined );
```

Comment:
@koke What about something like this? By adding https://github.com/WordPress/gutenberg/pull/19398/files#diff-6a57f8efae64419f8dbc4559ac9d5d9bR39 setting both the title and the template at the same time to prevent **two** un-dos

Reply:
That seems like a good solution, although I'm not sure about the new action and its name. Maybe we can call `editPost` directly from here with the title and content?

Reply:
Oh, I like that! I'll change it =)

Reply:
Changed 👍